apps/condo/domains/organization/schema/OrganizationEmployeeRole.js

```diff
@@ -24,15 +111,86 @@ const OrganizationEmployeeRole = new GQLListSchema('OrganizationEmployeeRole', {
         // There is no `user` reference, because Organization will have a set of pre-defined roles
         // and each employee will be associated with one of the role, not role with user.
 
+        isDefault: {
+            schemaDoc: '(Read-only) Indicates whether the role was added by default when the organization was created.' +
+                '\nSuch roles cannot be deleted and their name, description and “ticketVisibilityType” cannot be changed.',
+            type: 'Checkbox',
+            defaultValue: false,
+            isRequired: true,
+            kmigratorOptions: { default: false },
+            access: writeOnlyServerSideFieldAccess,
+        },
+        isEditable: {
+            schemaDoc: '(Read-only) Indicates whether the role can be edited',
+            type: 'Checkbox',
+            defaultValue: true,
+            isRequired: true,
+            kmigratorOptions: { default: true },
+            access: writeOnlyServerSideFieldAccess,
+        },
+
         name: {
+            schemaDoc: 'Role name. Cannot be changed for default roles',
             type: LocalizedText,
             isRequired: true,
             template: 'employee.role.*.name',
+            hooks: {
+                resolveInput: async ({ resolvedData, fieldPath, existingItem }) => {
+                    const newItem = { ...existingItem, ...resolvedData }
+                    const isDefaultRole = get(newItem, 'isDefault', false)
+                    if (isDefaultRole) return resolvedData[fieldPath]
+                    if (resolvedData[fieldPath] === undefined) return resolvedData[fieldPath]
+                    return normalizeText(resolvedData[fieldPath]) || ''
+                },
+                validateInput: async ({ resolvedData, fieldPath, context, existingItem, operation }) => {
+                    const newItem = { ...existingItem, ...resolvedData }
+                    const isUpdateOperation = operation === 'update'
+                    const isDefaultRole = get(newItem, 'isDefault', false)
+                    const name = get(resolvedData, fieldPath) || ''
+
+                    if (isUpdateOperation && isDefaultRole) {
+                        throw new GQLError(ERRORS.CANNOT_UPDATE_FIELD_FOR_DEFAULT_ROLE(fieldPath), context)
+                    }
```

Comment:
Admin and support cant edit default roles? Is it OK?

Reply:
It's OK
Default roles can be edited, except for the fields `name`, `description` and `ticketVisibilityType`. 
These fields cannot be edited by the admin and support too